internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -701,3 +722,34 @@ func handleUpdateError(err error) (bool, ctrl.Result, error) {
 	return true, ctrl.Result{Requeue: true, RequeueAfter: defaultRequeueAfterOnError},
 		errors.Wrap(err, "failed to update clusterDeployment")
 }
+
+func hasFailedToProvision(clusterDeployment *hivev1.ClusterDeployment, cluster *common.Cluster) bool {
+	conditions := clusterDeployment.Status.Conditions
+	conditionIdx := findConditionIndexByReason(AgentPlatformState, &conditions)
+	if conditionIdx == -1 {
```

Comment:
Not sure I understand why you need to check the CR conditions if you have the backend cluster